src/main/scala/org/scalajs/tools/tsimporter/Importer.scala

```diff
@@ -282,7 +282,15 @@ class Importer(val output: java.io.PrintWriter) {
         }
 
       case UnionType(left, right) =>
-        TypeRef.Union(typeToScala(left), typeToScala(right))
+        def visit(tpe: TypeTree, visited: List[TypeRef]): List[TypeRef] =
+          tpe match {
+            case UnionType(left, right) =>
+              visit(right, visit(left, visited))
+            case _ =>
+              visited ++ List(typeToScala(tpe))
```

Comment:
If you visit `right` before `left`, you can then do `typeToScala(tpe) :: visited` in this case.

Reply:
Good idea